### PR TITLE
[ENG-2276, ENG-2247] Add assetStatus and hydration to /search

### PR DESCRIPTION
**Pull request by rickliao**
Add assetStatus and hydration to /search

### CODE REVIEW

File: src/main/resources/world/data/api/swagger.json [hunk 1]
```diff
@@ -2977,99 +3013,82 @@
...
-              "comment",
-              "dataset",
-              "datatable",
-              "file",
-              "insight",
-              "integration",
-              "project",
-              "query"
-            ]
+    "SearchHydrationsPublic": {
```
Comment (rickliao) [inline src/main/resources/world/data/api/swagger.json:3016]: Swagger is confused about `SearchHydration` in public and private api. In swagger.json, it only has public-api `SearchHydration`. In reality, `PaginatedSearchResultsDto` for /collection/{owner} uses the private-api `SearchHydration`. In the end, I created a `SearchHydrationPublic` to hold the public-api one.

File: src/main/resources/world/data/api/swagger.json [hunk 2]
```diff
@@ -2977,99 +3013,82 @@
...
+      "properties" : {
+        "query" : {
+          "type" : "string"
         },
-        "resourceId": {
-          "type": "array",
-          "description": "Filter by resource IDs.",
-          "uniqueItems": true,
-          "items": {
-            "type": "string"
```
Comment (rickliao) [inline src/main/resources/world/data/api/swagger.json:3018]: We deleted this some time ago.

**Review by lydiaguarino — CHANGES_REQUESTED** [review]
Could you look at the whitespace inconsistencies here? It looks like we've removed some and added some - It looks like the general convention otherwise is NO space before the :

**Comment by rickliao** [comment]
ah yes, the swagger file adds space before `:`. Can I just run the whole thing through a json formatter?

**Review by lydiaguarino — APPROVED** [review]

**Comment by lydiaguarino** [comment]
Removing links to tickets, since this is a public repo :)

**Pull request merged**